File: 28-implement-strstr/28-implement-strstr.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        int idx,j;
        if(needle.length()==0)
            return 0;
        
        int len = haystack.length();
        
        for(int i = 0 ;i < len ; i++)
        {
            if(haystack[i]==needle[0])
            {
                 j=0;
                while(needle[j]==haystack[i+j])
                {
                    j++;
                    if(j==needle.length())
                        return i;
                    if(j>=len)
                        return -1;
                    
                }
                
            }
        }
        return -1;
    }
};

File: 28-implement-strstr/28-implement-strstr.java
class Solution {
    public int strStr(String haystack, String needle) {
        if(needle.length() == 0) return 0;
        for(int i = 0;i < haystack.length();i++){
            if(haystack.charAt(i) == needle.charAt(0)){
                int j = 0;
                while(j < needle.length() && i + j < haystack.length() && haystack.charAt(i + j) == needle.charAt(j))
                    j++;
                if(j == needle.length()) return i;
            }
        }
        return -1;
    }
}